### ISSUE TITLE
do not hard code and change code name...

### ISSUE THREAD

**Issue by proseLA** [zencart/zencart#4496]
https://github.com/zencart/zencart/blob/2aae461926f1082dc33cd6411229d17484ebe1cb/admin/modules.php#L249

total respect for @Zen4All but i hate this line of code.  as well as anything hard coded that makes it more difficult to troubleshoot.  if we are going to display the `$this->code`, then lets display it.  3 modules should not have the same code name.  it makes no sense to me.

**Comment by Zen4All** [comment]
Feel free to change it ;) All I did was updating the layout. I left all the other code as it was. And you are right, this should not be hard coded as it is now IMHO.

**Comment by proseLA** [comment]
@Zen4All you are right.  i should have dug further down.  apologies.  that line goes back to the original commit 10 years ago.

**Comment by Zen4All** [comment]
😁

**Comment by mc12345678** [comment]
Two things of note, one the only recommended "version" of this software is PayPal Express. The other, what information presented to shoppers/store owners would mean anything more to them and wouldn't presenting anything other than the text as "PayPal" be some sort of developer's violation? When I followed links within this site associated with branding requirements: https://developer.paypal.com/reference/guidelines/policiesAndGuidelines/ which led to Chapter 2 (see page 2.4 or pdf page 20 of 39) of https://www.paypalobjects.com/webstatic/en_US/developer/docs/pdf/PayPalX_Interface_Guide_Edition1_1.pdf, it appears that such "hard-coded" text is required, whether hard-coded there or elsewhere. Further on that, generally speaking the three modules are designed to basically only have one active at a time, so what troubleshooting is/was occurring that led down an unknown path?

**Comment by proseLA** [comment]
mc,
i am aware which version is the recommended version.  the modules page on the admin side of ZC, in its current form, does not tell me which script to look at when debugging any of the 3 modules.  so i have to open all 3, actually 6 when you count the languages files, to see which files go to which version.

i have also looked at said page in the linked pdf.  in no way shape or form does that page require an untitled column that refers to the name of the code in a table on an admin page in ZC to obliterate the name of the code to `PayPal`.  but thank you to pointing me to a 12 year old document that perhaps gives a bit of context to said line of code.  while admirably trying to conform to the wishes of paypal, it becomes a waste of time for developers like myself to always have to dig further because of some obscure reading of a branding document for customers.

**Comment by mc12345678** [comment]
> mc, i am aware which version is the recommended version. the modules page on the admin side of ZC, in its current form, does not tell me which script to look at when debugging any of the 3 modules. so i have to open all 3, actually 6 when you count the languages files, to see which files go to which version.
> 
> i have also looked at said page in the linked pdf. in no way shape or form does that page require an untitled column that refers to the name of the code in a table on an admin page in ZC to obliterate the name of the code to `PayPal`. but thank you to pointing me to a 12 year old document that perhaps gives a bit of context to said line of code. while admirably trying to conform to the wishes of paypal, it becomes a waste of time for developers like myself to always have to dig further because of some obscure reading of a branding document for customers.

I haven't been and still not entirely sure how to respond to the above emotionally charged response. I'll take my best shot in hopes not to cause significantly more aggravation.

I don't share the same inability to identify which payment module is associated with which line within the payment modules screen. The name(s) of the language files are a one-for-one to the module's filename. The content within the language file identifies the "full text" description of the payment module (which is shown in and searchable from the admin). Further, from even my initial experience of implementing PayPal processing, it was never permitted to activate all three modules at the same time, which makes it even a little more confusing about what the original issue was that initiated this thread. 

Do I not share the same trouble because I perhaps have looked at these modules a number of times over the years; that the modules tend to be displayed both in the admin and file list "alphabetically" by module filename so that the first, second and third of such files align; or is it because of the easy use of the developers tool kit from within the admin where one may ***assume*** access is/was available since the modules window is/was accessible. I don't know how I came to be familiar.

As to the "age" of one of the referenced links, I don't see how that has any bearing on the conversation when that particular document is/was referenced by related developer instruction that had been updated just a few months ago in October 2021.   There is a lot of discussion about what is and is not permitted to be in the program provided by a developer to a customer. There is not detailed discussion of what portion of the software is specifically the "customer" portion. In some cases/places it is clear that it is related to one making some sort of routine payment, but in other statements the discussion indicates that a customer could be all users of Zen Cart. Even if such users/admin at first start as non-paying customers of the software. Too literal? Maybe, but which is better, providing software that meets and exceeds the requirements or one that could be called to question and in some way restricted just because it is associated with a particular platform?

In short, I don't see the need to make the change discussed. It seems more like making a change just to make a change to the existing files. From what *I* need to use (perspective right?) I don't need two of three modules and in general they cause more trouble to maintain. But... They also have their place and purpose, so just because I don't need two of them doesn't make them unnecessary.

**Comment by proseLA** [comment]
i disagree with everything said.  everyone is not as intimately familiar with the code as others might be.

as far as never permitted, live site screenshot here:
![image](https://user-images.githubusercontent.com/1095136/147831799-5f1dbf49-0171-4cf5-934a-153381ff86db.png)
one line of code i recommended changing.

if members of this community want to be more beholden to a company whose complaints against them are numerous and far reaching, than an active supporter of this project; i suppose they are free to do so. (https://www.ftc.gov/system/files/documents/cases/1623102_c-4651_paypal_venmo_complaint_final.pdf)

this line does not go over and beyond the requirements of that document.  i feel it's a lame interpretation of that document that is wholly outside of the intention of said document.

**Comment by scottcwilson** [comment]
PR please.  I have always found this incredibly annoying.

**Comment by proseLA** [comment]
great minds think alike....  not sure what our excuse is....
